### PR TITLE
Attempt to fix Jackson Optional

### PR DESCRIPTION
**Pull request by claraadora**

### CODE REVIEW

**Review by gabztcr — APPROVED** [review]
LGTM. Keyword _looks_.

**Comment by gabztcr** [comment]
Tried and it doesn't show warnings anymore. @chrystalquek may want to comment on other side effects like absolute paths and test cases

File: build.gradle [hunk 1]
```diff
@@ -58,7 +58,9 @@ dependencies {
     implementation group: 'org.openjfx', name: 'javafx-graphics', version: javaFxVersion, classifier: 'linux'
 
     implementation group: 'com.fasterxml.jackson.core', name: 'jackson-databind', version: '2.7.0'
+    implementation group: 'com.fasterxml.jackson.core', name: 'jackson-core', version: '2.11.1'
```
Comment (chrystalquek) [inline build.gradle:61]: do we need to get approval adding dependencies heh

File: src/test/java/seedu/address/commons/util/ConfigUtilTest.java [hunk 2]
```diff
@@ -92,13 +92,22 @@ public void saveConfig_allInOrder_success() throws DataConversionException, IOEx
         //Try writing when the file doesn't exist
         ConfigUtil.saveConfig(original, configFilePath);
         Config readBack = ConfigUtil.readConfig(configFilePath).get();
-        assertEquals(original, readBack);
-
+        assertEquals(original.getUserPrefsFilePath().toAbsolutePath(),
+                readBack.getUserPrefsFilePath().toAbsolutePath());
+        System.out.println(original);
+        System.out.println(readBack);
+        System.out.println(original.getUserPrefsFilePath().toAbsolutePath());
```
Comment (chrystalquek) [inline src/test/java/seedu/address/commons/util/ConfigUtilTest.java:99]: as discussed, to test the absolute vs dynamic path, copy .jar file into empty folder and see whether the window size is saved

Comment (chrystalquek) [reply src/test/java/seedu/address/commons/util/ConfigUtilTest.java:99]: if its saved, i think we r quite safe

File: src/test/java/seedu/address/storage/StoragePersonManagerTest.java [hunk 3]
```diff
@@ -46,7 +46,8 @@ public void prefsReadSave() throws Exception {
         original.setGuiSettings(new GuiSettings(300, 600, 4, 6));
         storageManager.saveUserPrefs(original);
         UserPrefs retrieved = storageManager.readUserPrefs().get();
-        assertEquals(original, retrieved);
+        assertEquals(original.getAddressBookFilePath().toAbsolutePath(),
```
Comment (chrystalquek) [inline src/test/java/seedu/address/storage/StoragePersonManagerTest.java:49]: same here, use .jar file to test

**Pull request closed without merging**